docs/design/sum_types.md

```diff
@@ -0,0 +1,197 @@
+# Sum types
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Overview](#overview)
+-   [`choice` declarations](#choice-declarations)
+-   [User-defined sum types](#user-defined-sum-types)
+-   [Alternatives considered](#alternatives-considered)
+-   [References](#references)
+
+<!-- tocstop -->
+
+## Overview
+
+In Carbon, a _sum type_ is a type whose values are grouped into several distinct
+named cases, called _alternatives_. A value of a sum type logically consists of
```

Comment:
What do you mean with "logically" here? Are there other words that could make the intent clearer?

Reply:
OK, switched to "notionally".